models/user/user.go

```diff
@@ -413,6 +414,41 @@ func (u *User) DisplayName() string {
 	return u.Name
 }
 
+var emailToReplacer = strings.NewReplacer(
+	"\n", "",
+	"\r", "",
+	"<", "",
+	">", "",
+	",", "",
+	":", "",
+	";", "",
+)
+
+var isAZ09 = regexp.MustCompile(`^[a-zA-Z0-9 ]+$`)
+
+// EmailTo returns full name and email.
+func (u *User) EmailTo() string {
+	sanitizedDisplayName := emailToReplacer.Replace(u.DisplayName())
+
+	// we don't deal with non ascii strings
+	if !isAZ09.Match([]byte(sanitizedDisplayName)) {
+		return u.Email
+	}
```

Comment:
As per https://stackoverflow.com/a/41909208/808699, mime encoding should work in `From`, so I assume it will work in `To` too, so the correct encoding is like this:

```
From: =?utf-8?b?U3Ryw6FuZ8OpIE7DpG3DqQ==?= <test@example.com>
```

Reply:
![image](https://github.com/go-gitea/gitea/assets/24977596/f673dedb-aa56-465f-bea0-522cafbf99d8)

yes that works nice!